guides/howto/create-json-rest-api.md

```diff
@@ -0,0 +1,216 @@
+
+For our Links REST API we will need a SQL Database to store the links, therefore `--no-ecto` cannot be used on our command, but we will want to use the `--no-assets`, `--no-gettext` and `--no-mailer` in addition to the always necessary `--no-html` and `--no-live`.
+
+Bear in mind that nothing stops you to have a backend that supports simultaneously the REST API and a Web App (HTML, assets, internationalization and sockets).
+
+
+## Creating the Phoenix Application
+
+To create a REST API we need to first create a Phoenix application without using the unnecessary generators.
+
+First, run this command:
+
+```console
+mix phx.new links --no-assets --no-html --no-gettext --no-live --no-mailer
+```
+
+> **NOTE:** If we needed to support internationalization then we wouldn't have used `--no-gettext` in the command, and if we needed to send emails we also had to remove `--no-mailer`.
+
+Reply `Y` when prompted for:
+
+```console
+Fetch and install dependencies? [Yn] Y
+```
+
+After all the dependencies have been fetched and compiled we will get our prompt back.
+
+Now, we need to go inside the `links` folder:
+
+```console
+cd links
+```
+
+Then we need to configure our database in `config/dev.exs` and run:
+
+```console
+mix ecto.create
+```
+
+Optionally, we can track our progress with Git in order to see the changes applied
+by the generators when creating the REST API.
+
+```console
+git init
+git add -A
+git commit -m 'Created the Phoenix application'
+```
+
+### Creating the JSON REST API
+
+Now, we can finally create the REST API with:
+
+```console
+mix phx.gen.json Urls Url urls link:string title:string
+```
+
+Next, we need to add the `/url` resource to our `:api` scope in `lib/links_web/router.ex`:
+
+```elixir
+scope "/api", LinksWeb do
+  pipe_through :api
+  resources "/urls", UrlController, except: [:new, :edit]
+end
+```
+
+Then we need to update our repository by running migrations:
+
+```console
+mix ecto.migrate
+```
+
+Finally, we can use Git to see what changes were made by the generators:
+
+```console
+git status -u
+```
+
+The output should look similar to this:
+
+```console
+On branch master
+Changes not staged for commit:
+  (use "git add <file>..." to update what will be committed)
+  (use "git restore <file>..." to discard changes in working directory)
+   modified:   lib/links_web/router.ex
+
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+   lib/links/urls.ex
+   lib/links/urls/url.ex
+   lib/links_web/controllers/fallback_controller.ex
+   lib/links_web/controllers/url_controller.ex
+   lib/links_web/views/changeset_view.ex
+   lib/links_web/views/url_view.ex
+   priv/repo/migrations/20220510194408_create_urls.exs
+   test/links/urls_test.exs
+   test/links_web/controllers/url_controller_test.exs
+   test/support/fixtures/urls_fixtures.ex
```

Comment:
This needs to be updated based on a new directory structure, see: [Phoenix 1.7.0-rc.0 is out!](https://elixirforum.com/t/phoenix-1-7-0-rc-0-is-out/51625?u=eiji) for more information.